index.js

```diff
@@ -327,65 +419,86 @@ async function addSharedPartToReconciliation(
 }
 
 /**
- * This function loops through all shared parts (config files) and tries to add the shared part to each reconciliation listed in 'used_in'. It will make a POST request to the API. If the ID of one of the templates it missing, it will try to fetch it first by making a GET request. In case of success, it will store the details in the corresponding config files.
+ * This function loops through all shared parts (config files) and tries to add the shared part to each template listed in 'used_in'. It will make a POST request to the API. If the ID of one of the templates it missing, it will try to fetch it first by making a GET request. In case of success, it will store the details in the corresponding config files.
  * @param {Number} firmId
  */
-async function addAllSharedPartsToAllReconciliation(firmId) {
+async function addAllSharedParts(firmId) {
   const sharedPartsArray = fsUtils.getAllTemplatesOfAType("sharedPart");
-  for (let sharedPartName of sharedPartsArray) {
+  for await (let sharedPartName of sharedPartsArray) {
```

Comment:
I think the for .. of loop is already synchronous, so the await keyword isn't necessary

Reply:
I think we actually needed. Not sure if both awaits :thinking:

Reply:
I did a test and I believe it's finishing the first loop and doing it in sequential order, so I don't think the await is necessary here; checking this documentation https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/for-await...of ; I believe it still works cause we're using it to loop over 'sync iterables', but we can do that as well with the normal one: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/for...of

Reply:
Removed